### DISCUSSION TITLE
Draw arrow on top of PathLayer

### DISCUSSION THREAD

**Discussion by xamix** [mapsforge/vtm#1011] [Q&A] [answered]
Hi,

I'm looking to do draw an arrow on top of the PathLayer in my VTM android application.
It's the same thing that @devemux86 have done in his Cruiser application [here](https://user-images.githubusercontent.com/3484020/213869107-d65189c9-cd02-48d8-acc0-60c256210868.jpg)
(Looking for the white arrow on top of his pink Path layer):

However I'm stuck because if I add a texture to my path overlay, then it seem that the line is not drawn anymore
How to achieve the exact same thing than in Cruiser app?

Do I need two path overlay (One simple and one with Texture) as it seem it was proposed on another issue?
How is it done in Cruiser?

Regards

**Comment by devemux86** [comment, marked as answer]
> Do I need two path overlay (One simple and one with Texture) as it seem it was proposed on another issue?

Need two path layers, one with an arrow texture and stipple configuration.
The arrow bitmap can have transparent margins like the [old oneway arrow](https://github.com/mapsforge/vtm/blob/78658679731cec41490e69299bb25e4c93a5d3ed/vtm-themes/resources/assets/patterns/oneway.svg).

See also the [LineTexActivity](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/LineTexActivity.java) example for an implementation.

**Comment by xamix** [comment]
Thank you very much for your reply

I tested to add texture on my path layer however on segment junction in my path, the texture is splitted/truncated:

![arrow](https://user-images.githubusercontent.com/9742044/215185859-ba7e0d73-032c-4f69-a174-11b5c7360699.jpg)


This is generated with the following line style:

```java

LineStyle.LineBuilder().reset()
                    .stippleColor(c)
                    .stipple(240)
                    .stippleWidth(20f)
                    .strokeWidth(120f)
                    .strokeColor(c)
                    .fixed(true)
                    .texture(tex)
                    .randomOffset(false)
                    .build()
```
How have you resolved this?
Regards

**Reply by devemux86** [reply]
Using small textures to avoid breaks on corners.
Textures appear per segment, so at connections there are such problems.

Similar issues also exist when using textures in the render themes.

**Reply by xamix** [reply]
Thank you for your help.

I will use your advice to create 2 Path layers then.

Thank you very much!
